cabal-testsuite/PackageTests/ConditionalAndImport/cabal.out

```diff
@@ -89,7 +89,13 @@ Configuration is affected by the following files:
 - noncyclical-same-filename-b.project
 - same-filename/noncyclical-same-filename-b.config
     imported by: noncyclical-same-filename-b.project
-Up to date
+Resolving dependencies...
+Build profile: -w ghc-<GHCVER> -O1
+In order, the following will be built:
+ - my-0.1 (lib:my) (configuration changed)
+Configuring my-0.1...
+Preprocessing library for my-0.1...
+Building library for my-0.1...
```

Comment:
The package configuration now includes the project file path:

```haskell
ElaboratedConfiguredPackage
    { elabUnitId = UnitId "my-0.1-inplace"
    -- ...
    , elabPkgSourceLocation = WithConstraintSource
        { constraintInner = LocalUnpackedPackage "<ROOT>/."
        , constraintSource = ConstraintSourceProjectConfig
            ( ProjectConfigPath
                ( "noncyclical-same-filename-a.project" :| [] )
            )
        }
    -- ...
    }
```

Therefore, when we rebuild with a different `--project-file`, it changes the configuration.

Reply:
Isn't that correct anyway? If I switch from the default development config to `release` or `validate` then it should rebuild eveything.

Reply:
Fixed in e180d685b33e7e3ebd12350f08c8953af1cf57e3

IMO it should only rebuild if switching the project file causes other parameters like GHC flags or dependencies to change.

Reply:
Thanks! That makes sense. I'm fine with either behavior, since this sounds like an edge case. It looks like the test now only reruns the dependency solver.